0.data_analysis/construct_phenotypic_expression_plate_4_fs_data/nbconverted/cross_corr_all_cellprofiler.py

```diff
@@ -0,0 +1,349 @@
+                & (platedf[construct_col] == construct)
+                & (platedf["Metadata_Well"] != well)
+                & ~(platedf["Metadata_Well"].isin(tried_wells))
+            ]
+
+        else:
+            # Cells that are used to perform cross correlation. Cells with different siRNA constructs, and the same siRNA concentration from different wells are considered. Cells that have already been correlated are not tracked, since groups are compared independently
+            other_welldf = platedf.loc[
+                (platedf[conc_col] == conc)
+                & (platedf[construct_col] != construct)
+                & (platedf["Metadata_Well"] != well)
+            ]
+
+        # If there are no cells from either of these groups to cross correlate, the next group is considered
+        if len(welldf) == 0 or len(other_welldf) == 0:
+            continue
+
+        # Get the indices corresponding to each cell for the two dataframes to compare, where the indices are also used to reference the columns
+        welldf_idx = welldf[not_meta_cols].index
+        other_welldf_idx = other_welldf[not_meta_cols].index
+
+        # Find the correlation coefficients from the indices
+        corr_df = allcorr.loc[welldf_idx, other_welldf_idx]
+
+        # Convert the correlation values and flatten them
+        corr_vec = corr_df.values.flatten()
+
+        # There shouldn't be nans, but remove numpy nans from each correlation vector if nans exist
+        nan_idx = np.isnan(corr_vec)
+        corr_vec = corr_vec[~nan_idx]
+
+        # Keep track of the wells used for the correlation
+        tried_wells.append(well)
+
+        # Record the correlations constructs and SiRNA concentrations
+        transwell[f"{construct}_{conc}"].append(corr_vec)
+
+    return transwell
+
+
+# ## Get the correlations for the same constructs
+
+# In[13]:
+
+
+same_transwell = get_correlations(same_construct=True)
+
+# Concatenate all correlation values in one numpy array for each group
+same_construct = {
+    sirna_group: np.concatenate(corr_vecs)
+    for sirna_group, corr_vecs in same_transwell.items()
+}
+
+
+# ## Get the correlations for different constructs
+
+# In[14]:
+
+
+diff_transwell = get_correlations(same_construct=False)
+
+# Concatenate all correlation values in one numpy array for each group
+diff_construct = {
+    sirna_group: np.concatenate(corr_vecs)
+    for sirna_group, corr_vecs in diff_transwell.items()
+}
+
+
+# ## Prepare data for plotting
+
+# In[15]:
+
+
+# Get unique siRNA construct concentrations
+pos_covar = platedf[conc_col].unique()
+
+# Get unique siRNA constructs
+pos_const = platedf[construct_col].unique()
+
+# Define the domain for plotting
+xs = np.linspace(-1, 1, 500)
+
+# Define correlation distributions in a dictionary
+corr_dists = {
+    "Within Well": inwell,
+    "Same Construct Across Wells": same_construct,
+    "Different Construct Across Wells": diff_construct,
+}
+
+
+# ## Plot PDF Curves
```

Comment:
This is my first time seeing these curves for differences in cross-correlation. Consider adding curve interpretation in a `README.md` or in the notebook before or after the plots.

Reply:
I might add this in the next pr.